src/structlog/_utils.py

```diff
@@ -27,3 +28,19 @@ def until_not_interrupted(f: Callable[..., Any], *args: Any, **kw: Any) -> Any:
             if e.args[0] == errno.EINTR:
                 continue
             raise
+
+
+def get_processname() -> str:
+    # based on code from
+    # https://github.com/python/cpython/blob/313f92a57bc3887026ec16adb536bb2b7580ce47/Lib/logging/__init__.py#L342-L352
+    processname = "MainProcess"
```

Comment:
I wonder if these are the correct semantics. Is in the case of failure this likely to be correct, or is it just confusing? Would it be better to say `"n/a"`?

Reply:
Would be better yes, changing it.

Reply:
Changed to return `"n/a"` if process name cannot be retrieved.